File: SixUiViews/app/src/main/java/cn/six/open/view/rv/OneBindingAdapter/OneBindingAdapter.java
package cn.six.open.view.rv.OneBindingAdapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.annotation.LayoutRes;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by songzhw on 2017-02-21
 */

public class OneBindingAdapter<T> extends RecyclerView.Adapter<OneBindingViewHolder> {

    private final Context ctx;
    private final int layoutId;
    private final int brId;
    private final List<T> data;

    public OneBindingAdapter(Context ctx, @LayoutRes int itemLayoutId, int brId, List<T> data) {
        this.ctx = ctx;
        this.layoutId = itemLayoutId;
        this.brId = brId;
        this.data = data;
    }

    @Override
    public OneBindingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new OneBindingViewHolder(View.inflate(ctx, layoutId, null));
    }

    @Override
    public void onBindViewHolder(OneBindingViewHolder holder, int position) {
        ViewDataBinding binding = DataBindingUtil.bind(holder.itemView);
        if(data != null && data.size() > position){
            T t = data.get(position);
            binding.setVariable(brId, t);
            binding.executePendingBindings();
        }
    }

    @Override
    public int getItemCount() {
        return data == null? 0 : data.size();
    }
}

File: bak/OneAdapter_DataBinding/app/src/main/java/ca/six/one/lib/simple/OneBindingAdapter.kt
package ca.six.one.lib.simple

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import ca.six.one.lib.RvBindingViewHolder

class OneBindingAdapter<T>(var layoutResId: Int, var bindingVariable: Int, var data: List<T>? = null)
    : RecyclerView.Adapter<RvBindingViewHolder>() {

    override fun getItemCount(): Int {
        return if (data == null) 0 else data!!.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RvBindingViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding: ViewDataBinding = DataBindingUtil.inflate(inflater, layoutResId, parent, false)
        return RvBindingViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RvBindingViewHolder, position: Int) {
        val item = data!!.get(position)
        holder.binding.setVariable(bindingVariable, item)
        holder.binding.executePendingBindings()
    }
}
